File: backEnd/src/main/java/com/ruligames/backend/entities/Game.java
package com.ruligames.backend.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Entity
@Table(name = "game")
@Builder
public class Game {
    @Id
    @GeneratedValue
    private int gameId;
    @Column(name = "name")
    private String name;
    @Column(name = "price")
    private float price;
    @Column(name = "publisher_id")
    private int publisherId;
    @Column(name = "genre")
    private  String genre;
}

File: frontEnd/lib/models/game.dart
class Game {
  int? gameId;
  String? name;
  double? price;
  int? publisherId;
  String? genre;

  Game({this.gameId,this.name,this.price,this.publisherId,this.genre});

  Game.fromJson(Map<String, dynamic> json) {
    gameId = json['gameId'];
    name = json['name'];
    price = json['price'];
    publisherId = json['publisherId'];
    genre = json['genre'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['gameId'] = this.gameId;
    data['name'] = this.name;
    data['price'] = this.price;
    data['publisherId'] = this.publisherId;
    data['genre'] = this.genre;
    return data;
  }
}
